src/main/java/de/dennisguse/opentracks/io/file/TrackFileFormat.java

```diff
@@ -64,16 +62,12 @@ public String getExtension() {
         }
     },
     KMZ_ONLY_TRACK {
-
         private static final boolean exportPhotos = false;
 
         @Override
-        public TrackWriter newTrackWriter(Context context) {
-            return new KmlTrackWriter(context, false, false, exportPhotos);
-        }
-
-        public TrackExporter newTrackExporter(Context context) {
-            return newKmzTrackExporter(context, this.newTrackWriter(context), exportPhotos);
+        public TrackExporter createTrackExporter(Context context) {
+            KMLTrackExporter exporter = new KMLTrackExporter(context, false, false, exportPhotos);
```

Comment:
You use KML for KMLTrackExporter and GPX for GPXTrackExporter but Kmz for KmzTrackExporter.
When you use KmzTrackExporter, do you wanted to use KMZTrackExporter?

Reply:
Yes, as the KMLTrackExporter is used to generate the doc.xml for the KMZ.
It is in the following line passed to a KmzTrackExporter, which uses it to create the file and then zipping everything together.